Formula/bazel.rb

```diff
@@ -15,6 +16,9 @@ class Bazel < Formula
 
   def install
     ENV["EMBED_LABEL"] = "#{version}-homebrew"
+    # Bazel has it owns sandbox, ensure it works.
+    ENV.delete("SDKROOT")
+    ENV["CC"] = `xcrun -f #{ENV["CC"]}`.strip
```

Comment:
> (btw, the environment SDKROOT is simply incorrect. I don't know who set it, I don't have it in my environment so it's coming from some part of Homebrew).

It's not necessary to set it in all cases, but it's far from being incorrect. Homebrew sets it in certain circumstances and tools like `xcrun` and a few others respect it. Also many build systems either pick it up or at least don't tamper with it. Check out the `xcrun(1)` man page for details.

> There is no way to tweak the action environment in Bazel (that would be against its philosophy).

Forgive me, but not having spent time studying or using Bazel, I can't really comment on this and don't understand what the “action environment” is. But this decision feels unfortunate.

> IMHO the real correct way would be for the wrapper in the superenv to actually includes those information and make themselves un-distinguishable from a normal compiler (so they don't need those HOMEBREW_* environment variables).

There are many ways for inter-process communication and I doubt there is the *one true way*. Environment variables are possibly one of the more convenient and efficient solution and have worked quite well for us. (And the “misbehaving” cases are usually caught by the check you've stumbled into.) We could use files or UNIX domain sockets to pass that information from the main process to our wrappers, but files are also not immune to tampering. (And we have to pass *some* information, so a communication channel is unavoidable.)

> Note that the previous version of bazel was directly coding the path to /usr/bin/gcc in its C++ configuration file […]

That's even worse. You realize that GCC is not the only C/C++ compiler suite, can be installed in many different locations and under different names, and specifically `gcc` is not even the right frontend if you're targeting C++ code? (Sorry for this mini-rant.)

To be honest, I'm not really happy with the situation and don't feel like we have found a good solution yet. But I'll ask other maintainers to have a look. I don't want to waste too much of your time if it turns out that I'm being overly stubborn and others find your hack/solution acceptable.